### PR TITLE
Add Payment Activity's empty state and filters [WIP]

### LINKED ISSUE #8198: Payments Overview Widget - Scaffolding and Empty State
### Description

Create the container component to house the other components - #8193, #8194, #8196, and #8197.

### Acceptance criteria

> [!TIP]
> Check the design in additional context section below for up-to-date requirements.

* Create the container component with the title "Your payment activity".
* When there's no data, show a filler banner (empty state).
* Add Tracks event(s) to record widget views.

### Designs

image

### Additional context

* Project - paJDYF-bWn-p2
* Designs - p6riRB-9OY-p2
* Widget breakdown - pepjwh-Jw-p2

### PR DESCRIPTION
**Pull request by elazzabi**
Fixes #8198 

#### Changes proposed in this Pull Request

- Introducing an empty state based on TPV for account

#### Testing instructions

- Get server PR 4775
- Open Payments > Overview
- You should see a new card for Payments Activity with "You have some money". This will be replaced with actual graphs
- You should see an empty state if you don't have a TPV in your account. If you want to force it, change `tpv === 0` to `tpv !== 0`

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8234` or branch name `add/payment-activity-empty-state` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/payment-activity-empty-state) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 8522eaf9153d4551afcb7c3d92cceba4bae31f3c
- Build time: 2024-03-11 17:02:35 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/payment-activity/index.tsx [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+import './style.scss';
+
+const PaymentActivity: React.FC = () => {
+	const {
+		accountDefaultCurrency,
+		transactions_data: { tpv },
+	} = wcpaySettings;
+	const [ before, setBefore ] = useState( moment() );
+	const [ after, setAfter ] = useState(
+		moment().subtract( 7, 'days' ).startOf( 'day' )
```
Comment (anu-rock) [inline client/components/payment-activity/index.tsx:78]: Is the idea to keep the number 7 always or read it from the filters when they are ready?

File: client/components/payment-activity/test/index.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+describe( 'PaymentActivity component', () => {
+	test( 'Component renders an empty state', () => {
+		global.wcpaySettings = {
+			transactions_data: {
+				tpv: 0,
+			},
+		};
+
+		const { container, getByText } = render( <PaymentActivity /> );
```
Comment (anu-rock) [inline client/components/payment-activity/test/index.tsx:28]: The `container` variable is unused here.

File: client/components/payment-activity/index.tsx [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+	);
+	const [ rangeLabel, setRangeLabel ] = useState( 'Last 7 days' );
```
Comment (anu-rock) [inline client/components/payment-activity/index.tsx]: Missing i18n translation? :D

File: client/components/payment-activity/index.tsx [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+
+	const afterText = after ? moment( after ).format( 'MM/DD/YYYY' ) : 'From';
```
Comment (anu-rock) [inline client/components/payment-activity/index.tsx]: `MM/DD/YYYY` is a US-specific format. I'm wondering if we should instead use a more standardized format here, like `YYYY-MM-DD`?

File: client/components/payment-activity/index.tsx [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+								aria-expanded={ isOpen }
+							>
+								{ rangeLabel }:
+								<span className="wcpay-payments-activity__label__span">
+									{ after.format( 'MMM D' ) } -{ ' ' }
+									{ before.format( 'MMM D, YYYY' ) }
+								</span>
+							</Button>
+						) }
+						renderContent={ () => (
```
Comment (anu-rock) [inline client/components/payment-activity/index.tsx]: Wow, you literally created the whole thing from scratch :D Kudos for this effort. I'm sure it will prove a great start when we separarely work on the [filters component](https://github.com/Automattic/woocommerce-payments/issues/8193), assuming we'll extract this code into a separate PR at that point.

File: client/components/payment-activity/style.scss [hunk 6]
```diff
@@ -0,0 +1,34 @@
+.wcpay-payments-activity {
+	&__card__body {
+		text-align: center;
+		padding: 32px 0 !important;
```
Comment (anu-rock) [inline client/components/payment-activity/style.scss]: I know we have a precendent for using `!important` elsewhere in the codebase, but I wonder if it's really required.

File: client/overview/index.js [hunk 7]
```diff
@@ -182,12 +183,24 @@ const OverviewPage = () => {
...
-							<Card>
-								<Welcome />
-								<AccountBalances />
-							</Card>
+							<>
+								<Card>
+									<Welcome />
+									<AccountBalances />
+								</Card>
+								<Card>
```
Comment (anu-rock) [inline client/overview/index.js]: Things have started to get repetitive in the if-else conditions. Time for a refactor?

File: includes/class-wc-payments-account.php [hunk 8]
```diff
@@ -2089,4 +2089,14 @@ private function get_onboarding_user_data(): array {
 			'referer'         => isset( $_SERVER['HTTP_REFERER'] ) ? esc_url_raw( wp_unslash( $_SERVER['HTTP_REFERER'] ) ) : '',
 		];
 	}
+
+	/**
+	 * Gets the current account fees for rendering on the settings page.
```
Comment (anu-rock) [inline includes/class-wc-payments-account.php]: The doc comments need to be updated as per the new method.

**Review by anu-rock — COMMENTED** [review]
Great work here 👏 
___

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | ✅ All good |
| Automated tests  | ✅ All good |
| Naming  | ✅ All good |
| Typing  | ✅ All good |
| Code comments  | 👍  Almost good (left a comment) |
| Changelog  | Skipped (until the PR is finalized) |
| Demo (video or JN site)  | Skipped (until the PR is finalized) |
| Manual testing  | ✅ Works as expected (tried the `tpv !==0` hack) |

File: client/components/payment-activity/index.tsx (hunk 3 above)
Comment (elazzabi) [reply client/components/payment-activity/index.tsx]: Haha, I catched this afterward, thank you for the review!

File: client/components/payment-activity/index.tsx (hunk 4 above)
Comment (elazzabi) [reply client/components/payment-activity/index.tsx]: I switched it to `DD/MM/YYYY` for now.

I also added a comment to explain that we hide this anyway (so what we display doesn't matter). I was hoping I could remove it altogether, but it seems the component depends on it to work, so keeping it there 🤷 

I also considered fixing this upsteam in the `@woocommerce/components` library, but it depends on the team continuing the work here :/

File: client/components/payment-activity/style.scss (hunk 6 above)
Comment (elazzabi) [reply client/components/payment-activity/style.scss]: It looks like we don't need this anymore, so I removed it altogether here 41228facd.

That said, I'm also using `!important` in other parts of the CSS because we can't access the component's styling. I was hoping to fix this upstream too...

File: includes/class-wc-payments-account.php (hunk 8 above)
Comment (elazzabi) [reply includes/class-wc-payments-account.php]: Oh yes, thanks! 639c072a0

File: client/components/payment-activity/test/index.tsx (hunk 2 above)
Comment (elazzabi) [reply client/components/payment-activity/test/index.tsx:28]: This has been used in the tests now

File: client/components/payment-activity/index.tsx (hunk 1 above)
Comment (elazzabi) [reply client/components/payment-activity/index.tsx:78]: The idea is the **default** stats will be for the past week, hence using the 7 here. Once the user interacts with the filters, this value will change accordingly.

**Comment by elazzabi** [comment]
Not including a changelog here as this will depend on the team taking this forward...

File: client/components/payment-activity/index.tsx (hunk 1 above)
Comment (anu-rock) [reply client/components/payment-activity/index.tsx:78]: Does it make sense to read this value always from the "compared to" filter? We can probably depend on 7 being the default value for that filter.

Comment (anu-rock) [reply client/components/payment-activity/index.tsx:78]: I am not against having a default value here, just against the idea of hard-coding. In that way, this component seems to be doing more than its paygrade.

Comment (elazzabi) [reply client/components/payment-activity/index.tsx:78]: I agree that we shouldn't hard-code things when it's not needed, but I don't think that depending on the "compared to" filter is the way to go.

I think the two should be separate or at least the other way around, "Compared to:" depending on the "From:" filter. Users will select a date from the "From:" filter first and expect to have the "Compared to:" automatically follow the time range.

However, I believe these will be more apparent when starting the "data connection" phase. So if I'm handing this over to another team, I prefer to keep it separate so they can work on their preferred logic afterward. Otherwise, I might be introducing an extra burden to understand my logic...

What do you think?

Comment (anu-rock) [reply client/components/payment-activity/index.tsx:78]: Yep, that makes sense 👍 WDYT about putting a TODO note about it in code and keeping this thread open (unresolved)?

Comment (elazzabi) [reply client/components/payment-activity/index.tsx:78]: Agree! Added a TODO and linked to this thread 2903a10c29ac6e1a9bb664dc58f35070220537db

**Comment by mordeth** [comment]
Excellent job, @elazzabi ! The code appears to be well done. Something I observed is that tracks events are not yet implemented for both the empty state and filter interactions. It would be beneficial to have them in accordance with the acceptance criteria.

**Comment by elazzabi** [comment]
Handover notes pepjwh-Ok-p2#comment-926

**Comment by nagpai** [comment]
Closing the PR since it was handed over and addressed via #8558
